src/types/object.js

```diff
@@ -35,4 +35,12 @@ export default parameterized(T => class ObjectType {
   delete(name) {
     return filter(({ key }) => key !== name, valueOf(this));
   }
+
+  map(fn) {
+    return map((v, key) => fn(sourceOf(this[key]), key), valueOf(this));
```

Comment:
It might be conceptually (and computationally) simpler to just create the mapping context using `create` directly instead of `sourceOf` like we do in the `ArrayType`.

if we access the substate via the parent state we're reifying the substate (via create), then mounting it, then getting the source of the mounted substate, and using it as the mapping context.

If on the other hand, we just create directly, we skip everything except the create portion.

Another key advantage is that the `sourceOf` API remains private to the metadata. Not only do we have to not commit to maintaining this API, but it means less concepts for authors who want to implement their own parameterized types.

```suggestion
    return map(value => valueOf(fn(create(T, value))), valueOf(this));
```

Of course, if the property already _is_ reified, then it's going to be fast. 

🤷‍♂️ Hard to say which is going to be more often the case without data, so I'd opt for simplicity here.

Reply:
Sounds good. By leveraging the public API, it is certainly a lot easier to understand what's going on there. I applied your suggested change.